### ISSUE TITLE
Only choosing #000000

### ISSUE THREAD

**Issue by versak** [claviska/jquery-minicolors#137]
I am using a simple chooser in my site and the slider and picker move as expected, but only ever choose #000000 as the value.

I'm using the default settings with only the "bootstrap" theme selected. The inputs are copied right from the sample html file included in the package.  I'm calling the script by ID and return (in the change function) #000000 in the console.

Any idea what I'm missing or doing wrong? The site is on an intranet host so I don't have a link. I'm using angularjs and bootstrap, and the page with the script is an included template html file.

thanks

**Comment by SolveSoul** [comment]
Hi, I'm having the exact same problem here with jQuery v3.1.1. As mentioned by @garlofrank replacing outerWidth() and outerHeight() by outerWidth(true) and outerHeight(true) in the source code fixes this issue.

The function where outerWidth and outerHeight can found is:
`function getCoords(input, target)`

Is there a specific reason this hasn't been implemented in the library yet?

**Comment by claviska** [comment]
> Is there a specific reason this hasn't been implemented in the library yet?

Please read above. I haven't been able to replicate it. One user says it affects jQuery 1.x. Another is saying 3.x. Nobody has posted a test case and everything works fine when when I test it.

Before I can fix a bug, I need to be able to replicate it to verify the issue is with the plugin itself and not something to do with custom scripts/styles interfering. I'm not going to jump through hoops to identify/fix bugs that nobody wants to post a test case for. 🤷‍♂️

**Comment by SolveSoul** [comment]
Fair enough. I'll try to share as much information as possible:

I serve the script via Gulp. I concatenate all my libraries and own JS files to one big file and minify it. The same goes for the CSS. To make sure this isn't an issue I've also tried serving both the JS and CSS separately. Unfortunately, this doesn't resolve the bug.

jQuery version: `3.1.1`
Bootstrap version: `3.3.7`

HTML:
`<input name="status-bar-color" type="text" class="minicolors" value="#ffffff" />`

JS:
```	
$(document).ready(function(){
	$('.minicolors').minicolors({
		theme: 'bootstrap',
		control: 'hue',
		change: function(color, opacity) {
			console.log(value + ' - ' + opacity);
		}
	});
});
```

It's a pretty standard set-up, let me know if I'm missing something then I can add it to the test case.

**Comment by claviska** [comment]
I took the time to create a minimal test case based on your example. MiniColors doesn't seem to work with any of the fiddle sites (need to look into that later), so I'm posting it here as an example:

```html
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>Test Case</title>
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
  <link rel="stylesheet" href="https://cdn.rawgit.com/claviska/jquery-minicolors/master/jquery.minicolors.css">
</head>
<body style="padding: 30px;">
  
  <div style="max-width: 300px;">
    <input name="status-bar-color" type="text" class="minicolors form-control" value="#ffffff" />
  </div>
  
  <script src="https://code.jquery.com/jquery-3.2.1.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/tether/1.4.0/js/tether.min.js"></script>
  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
  <script src="https://cdn.rawgit.com/claviska/jquery-minicolors/master/jquery.minicolors.js"></script>
  <script src="https://cdn.rawgit.com/claviska/jquery-minicolors/master/jquery.minicolors.js"></script>
  
  <script>
    $(document).ready(function(){
    	$('.minicolors').minicolors({
    		theme: 'bootstrap',
    		control: 'hue'
    	});
    });
  </script>  
</body>
</html>
```

Simply load that file in your browser to run it. Alas, I still can't reproduce the issue (don't you love the 256 color version of the palette?):

![2017-03-23 09 46 43](https://cloud.githubusercontent.com/assets/55639/24250575/bcee962c-0fad-11e7-8398-14d9b9d3663b.gif)

This suggests there's an issue with the implementation, not the plugin itself.

I'd recommend reducing your code to determine where the conflict is. If we identify an edge case, we can work on a patch. However, you may even realize an issue in your own code causing the conflict which will be even easier to fix.

**Comment by SolveSoul** [comment]
Thanks for looking into this in-depth. I've copied your code into an HTML file and it works immediately.

I've tried excluding other dependencies one by one in my project and found the solution when I excluded `jquery-ui-1.8.5.custom.min.js`. When this file is excluded the color picker seems to be working correctly. Since the name of this script contains 'custom' I suspect one of the previous developers tampered with the jQuery UI plugin (unless there is some custom jQuery UI version out there, a quick Google search didn't yield any results, though). If I try the default jQuery UI 1.8.5 it also seems to work correctly (`http://ajax.googleapis.com/ajax/libs/jqueryui/1.8.5/jquery-ui.min.js`). My guess is that the custom script overwrote some functionality of jQuery.

So basically, some weird version of jQuery UI was causing this issue.

Anyway, thanks a lot!
